apps/backend/src/modules/auth/auth.service.ts

```diff
@@ -21,6 +25,57 @@ class AuthService {
 		this.encrypt = encrypt;
 	}
 
+	public async forgotPassword(payload: EmailDto): Promise<null> {
+		const { email: targetEmail } = payload;
+
+		const user = await this.userService.findByEmail(targetEmail);
+
+		if (!user) {
+			throw new AuthError({
+				message: ErrorMessage.INCORRECT_CREDENTIALS,
+				status: HTTPCode.UNAUTHORIZED,
+			});
+		}
+
+		const userDetails = user.toObject();
+
+		const jwtToken = await token.createToken({
+			userId: userDetails.id,
+		});
+
+		const link = `http://${config.ENV.APP.HOST}:3000/reset-password/${jwtToken}`;
+
+		mail.sendResetPasswordEmail({
+			recipient: userDetails.email,
+			resetLink: link,
+		});
+
+		return await Promise.resolve(null);
```

Comment:
await is redundant here

Reply:
And you can just return null, async function automatically wraps returned result with Promise. Please, check other places in your code to return just null in async functions